src/realm/sync/noinst/pending_bootstrap_store.cpp

```diff
@@ -206,27 +209,19 @@ bool PendingBootstrapStore::has_pending() const noexcept
     return m_has_pending;
 }
 
-void PendingBootstrapStore::clear()
-{
-    auto tr = m_db->start_write();
-    clear(*tr);
-    tr->commit();
-}
-
-void PendingBootstrapStore::clear(Transaction& wt)
+void PendingBootstrapStore::clear(Transaction& wt, int64_t query_version)
 {
     auto bootstrap_table = wt.get_table(m_table);
     bootstrap_table->clear();
+    if (m_has_pending) {
+        m_subscription_store->cancel_bootstrap(wt, query_version);
+    }
     m_has_pending = false;
 }
 
-PendingBootstrapStore::PendingBatch PendingBootstrapStore::peek_pending(size_t limit_in_bytes)
+PendingBootstrapStore::PendingBatch PendingBootstrapStore::peek_pending(Transaction& tr, size_t limit_in_bytes)
 {
-    auto tr = m_db->start_read();
-    auto bootstrap_table = tr->get_table(m_table);
-    if (bootstrap_table->is_empty()) {
```

Comment:
peek_pending() and pop_front_pending() are only called if has_pending() returns true, so the table should never be empty in either of them.